axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceFinancialDiscountServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.axelor.apps.account.service.invoice;
+
+import com.axelor.apps.account.db.FinancialDiscount;
+import com.axelor.apps.account.db.Invoice;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public class InvoiceFinancialDiscountServiceImpl implements InvoiceFinancialDiscountService {
+
+  protected InvoiceService invoiceService;
+
+  @Inject
+  public InvoiceFinancialDiscountServiceImpl(InvoiceService invoiceService) {
+    this.invoiceService = invoiceService;
+  }
+
+  @Override
+  public void setFinancialDiscountInformations(Invoice invoice) {
+
+    Objects.requireNonNull(invoice);
+
+    if (invoice.getFinancialDiscount() != null) {
+      FinancialDiscount financialDiscount = invoice.getFinancialDiscount();
+      invoice.setLegalNotice(financialDiscount.getLegalNotice());
+      invoice.setFinancialDiscountRate(financialDiscount.getDiscountRate());
+      invoice.setFinancialDiscountTotalAmount(
+          computeFinancialDiscountTotalAmount(invoice, financialDiscount));
+      invoice.setRemainingAmountAfterFinDiscount(
+          invoice.getInTaxTotal().subtract(invoice.getFinancialDiscountTotalAmount()));
+
+      if (invoice.getDueDate() != null) {
+        invoice.setFinancialDiscountDeadlineDate(
+            invoiceService.getFinancialDiscountDeadlineDate(invoice, financialDiscount));
+      }
+    } else {
+      resetFinancialDiscountInformations(invoice);
+    }
+  }
+
+  protected BigDecimal computeFinancialDiscountTotalAmount(
+      Invoice invoice, FinancialDiscount financialDiscount) {
+    return financialDiscount
+        .getDiscountRate()
+        .multiply(invoice.getInTaxTotal())
+        .divide(new BigDecimal(100));
```

Comment:
It is not a huge issue especially in the current code, but it might be better to set a scale here (equal to the scale of the field `financialDiscountTotalAmount`